### PR TITLE
Pass Release Version only if not None

### PR DESCRIPTION
**Pull request by sthirugn**

### CODE REVIEW

**Comment by sthirugn** [comment]
```py
# Test scenario 1: enable with release_ver=none
>>> entities.Product(id=prd_id).enable_rhrepo(base_arch='x86_64',release_ver=None,reposet_id=reposet_id)
{u'username': u'admin', u'ended_at': u'2015-07-30T16:26:04Z', u'output': {}, u'cli_example': None, u'label': u'Actions::Katello::RepositorySet::EnableRepository', u'state': u'stopped', u'result': u'success', u'input': {u'product': {u'label': u'Red_Hat_Enterprise_Linux_Server', u'id': 113, u'cp_id': u'69', u'name': u'Red Hat Enterprise Linux Server'}, u'repository': {u'label': u'Red_Hat_Satellite_Tools_6_Beta_for_RHEL_7_Server_RPMs_x86_64', u'id': 12, u'name': u'Red Hat Satellite Tools 6 Beta for RHEL 7 Server RPMs x86_64'}, u'locale': u'en', u'provider': {u'id': 6, u'name': u'Red Hat'}, u'organization': {u'label': u'HqUqxzbg', u'id': 4, u'name': u'HqUqxzbg'}, u'services_checked': [u'pulp', u'pulp_auth', u'candlepin', u'candlepin_auth', u'elasticsearch']}, u'progress': 1.0, u'started_at': u'2015-07-30T16:25:57Z', u'humanized': {u'action': u'Enable', u'input': [[u'repository', {u'text': u"repository 'Red Hat Satellite Tools 6 Beta for RHEL 7 Server RPMs x86_64'", u'link': None}], [u'product', {u'text': u"product 'Red Hat Enterprise Linux Server'", u'link': u'#/products/113/info'}], [u'organization', {u'text': u"organization 'HqUqxzbg'", u'link': u'/organizations/4/edit'}]], u'errors': [], u'output': u''}, u'id': u'b420bdb4-501b-4ed1-9380-364485c408d5', u'pending': False}

# Test scenario 2: disable with release_ver=none
>>> entities.Product(id=prd_id).disable_rhrep(base_arch='x86_64',release_ver=None,reposet_id=reposet_id)
{u'username': u'admin', u'ended_at': u'2015-07-30T16:29:11Z', u'output': {}, u'cli_example': None, u'label': u'Actions::Katello::RepositorySet::DisableRepository', u'state': u'stopped', u'result': u'success', u'input': {u'product': {u'label': u'Red_Hat_Enterprise_Linux_Server', u'id': 113, u'cp_id': u'69', u'name': u'Red Hat Enterprise Linux Server'}, u'repository': {u'label': u'Red_Hat_Satellite_Tools_6_Beta_for_RHEL_7_Server_RPMs_x86_64', u'id': 12, u'name': u'Red Hat Satellite Tools 6 Beta for RHEL 7 Server RPMs x86_64'}, u'locale': u'en', u'provider': {u'id': 6, u'name': u'Red Hat'}, u'organization': {u'label': u'HqUqxzbg', u'id': 4, u'name': u'HqUqxzbg'}, u'services_checked': [u'elasticsearch', u'pulp', u'pulp_auth', u'candlepin', u'candlepin_auth']}, u'progress': 1.0, u'started_at': u'2015-07-30T16:29:09Z', u'humanized': {u'action': u'Disable', u'input': [[u'repository', {u'text': u"repository 'Red Hat Satellite Tools 6 Beta for RHEL 7 Server RPMs x86_64'", u'link': None}], [u'product', {u'text': u"product 'Red Hat Enterprise Linux Server'", u'link': u'#/products/113/info'}], [u'organization', {u'text': u"organization 'HqUqxzbg'", u'link': u'/organizations/4/edit'}]], u'errors': [], u'output': u''}, u'id': u'fafde1db-4200-46cd-8696-579815452b8b', u'pending': False}



# Test scenario 3: enable with valid release_ver
>>> entities.Product(id=prd_id).enable_rhrepo(base_arch='x86_64',release_ver=u'6.7',reposet_id=reposet_id)
{u'username': u'admin', u'ended_at': u'2015-07-30T16:34:31Z', u'output': {}, u'cli_example': None, u'label': u'Actions::Katello::RepositorySet::EnableRepository', u'state': u'stopped', u'result': u'success', u'input': {u'product': {u'label': u'Red_Hat_Enterprise_Linux_Server', u'id': 113, u'cp_id': u'69', u'name': u'Red Hat Enterprise Linux Server'}, u'repository': {u'label': u'Red_Hat_Enterprise_Virtualization_Agents_for_RHEL_6_Server_Beta_RPMs_x86_64_6_7', u'id': 13, u'name': u'Red Hat Enterprise Virtualization Agents for RHEL 6 Server Beta RPMs x86_64 6.7'}, u'locale': u'en', u'provider': {u'id': 6, u'name': u'Red Hat'}, u'organization': {u'label': u'HqUqxzbg', u'id': 4, u'name': u'HqUqxzbg'}, u'services_checked': [u'pulp', u'pulp_auth', u'candlepin', u'candlepin_auth', u'elasticsearch']}, u'progress': 1.0, u'started_at': u'2015-07-30T16:34:22Z', u'humanized': {u'action': u'Enable', u'input': [[u'repository', {u'text': u"repository 'Red Hat Enterprise Virtualization Agents for RHEL 6 Server Beta RPMs x86_64 6.7'", u'link': None}], [u'product', {u'text': u"product 'Red Hat Enterprise Linux Server'", u'link': u'#/products/113/info'}], [u'organization', {u'text': u"organization 'HqUqxzbg'", u'link': u'/organizations/4/edit'}]], u'errors': [], u'output': u''}, u'id': u'b4aec48e-17c6-4374-8a88-dfb46f52640c', u'pending': False}

# Test scenario 4: enable with valid release_ver
>>> entities.Product(id=prd_id).disable_rhrepo(base_arch='x86_64',release_ver=u'6.7',reposet_id=reposet_id)
{u'username': u'admin', u'ended_at': u'2015-07-30T16:36:36Z', u'output': {}, u'cli_example': None, u'label': u'Actions::Katello::RepositorySet::DisableRepository', u'state': u'stopped', u'result': u'success', u'input': {u'product': {u'label': u'Red_Hat_Enterprise_Linux_Server', u'id': 113, u'cp_id': u'69', u'name': u'Red Hat Enterprise Linux Server'}, u'repository': {u'label': u'Red_Hat_Enterprise_Virtualization_Agents_for_RHEL_6_Server_Beta_RPMs_x86_64_6_7', u'id': 13, u'name': u'Red Hat Enterprise Virtualization Agents for RHEL 6 Server Beta RPMs x86_64 6.7'}, u'locale': u'en', u'provider': {u'id': 6, u'name': u'Red Hat'}, u'organization': {u'label': u'HqUqxzbg', u'id': 4, u'name': u'HqUqxzbg'}, u'services_checked': [u'elasticsearch', u'pulp', u'pulp_auth', u'candlepin', u'candlepin_auth']}, u'progress': 1.0, u'started_at': u'2015-07-30T16:36:34Z', u'humanized': {u'action': u'Disable', u'input': [[u'repository', {u'text': u"repository 'Red Hat Enterprise Virtualization Agents for RHEL 6 Server Beta RPMs x86_64 6.7'", u'link': None}], [u'product', {u'text': u"product 'Red Hat Enterprise Linux Server'", u'link': u'#/products/113/info'}], [u'organization', {u'text': u"organization 'HqUqxzbg'", u'link': u'/organizations/4/edit'}]], u'errors': [], u'output': u''}, u'id': u'e1d2f477-c0f7-4957-8196-1185371e5915', u'pending': False}
```

**Comment by oshtaier** [comment]
Just in case, we are moving functions mentioned in your example to separate entity:
https://github.com/SatelliteQE/nailgun/pull/166

**Comment by sthirugn** [comment]
@oshtaier I need this PR to retrigger automation.  May be we can merge this and you all can rebase https://github.com/SatelliteQE/nailgun/pull/166 after that?

File: nailgun/entities.py [hunk 1]
```diff
@@ -2702,9 +2702,13 @@ def enable_rhrepo(self, base_arch,
             Return JSON response otherwise.
 
         """
+        data = {}
+        data[u'basearch'] = base_arch
```
Comment (JacobCallahan) [inline nailgun/entities.py]: do we need to check if base_arch is not None as well?

**Comment by omaciel** [comment]
@sthirugn I recommend that we get #165 and #166 merged and close this one. I understand we want to run the automation soon-ish but let's not rush this refactoring :)

File: nailgun/entities.py (hunk 1 above)
Comment (sthirugn) [inline nailgun/entities.py]: I don't think there is a need - base_arch is a must for all repos - so the calling code should make sure of that.  
With recent change in sat, release version is made optional and should not be passed with invalid or empty values.

Comment (JacobCallahan) [inline nailgun/entities.py]: :+1:

**Comment by sthirugn** [comment]
Closed in favor of https://github.com/SatelliteQE/nailgun/pull/166

**Pull request closed without merging**